Formula/ruby.rb

```diff
@@ -64,6 +71,12 @@ def install
     #       https://github.com/Homebrew/brew/pull/12508
     inreplace "tool/mkconfig.rb", /^(\s+val = )'"\$\(SDKROOT\)"'\+/, "\\1"
 
+    # The Homebrew GCC shim does not accept CFLAGS `-std=c99`.
+    if OS.linux?
+      inreplace ".bundle/gems/rbs-2.7.0/ext/rbs_extension/extconf.rb",
+                "-std=c99", "-std=gnu99"
```

Comment:
I don't think we strip `-std` so this doesn't seem right.

Reply:
The `./configure` command detects compiler option compatibility. For GCC C99, the available option is `-std=gnu99` rather than `-std=c99`. The `./configure` will detect and use `-std=gnu99` to compile `ruby` and `libruby.so`.

For gems, they do not have a `./configure` command. In `rbs-2.7.0/ext/rbs_extension/extconf.rb`, it arbitrarily adds `-std=c99` to CFLAGS and does not check it. GCC does not recognize option `-std=c99`. We can either inreplace it with `-std=gnu99`, or just comment out the whole line.

```ruby
# .bundle/gems/rbs-2.7.0/ext/rbs_extension/extconf.rb"

require 'mkmf'
$INCFLAGS << " -I$(top_srcdir)" if $extmk
append_cflags ['-std=c99']
create_makefile 'rbs_extension'
```

Reply:
> A new edition of the ISO C standard was published in 1999 as ISO/IEC 9899:1999, and is commonly known as C99. (While in development, drafts of this standard version were referred to as C9X.) GCC has substantially complete support for this standard version; see https://gcc.gnu.org/c99status.html for details. To select this standard, use -std=c99 or -std=iso9899:1999.

https://gcc.gnu.org/onlinedocs/gcc/Standards.html

GCC accepts `-std=c99`.